### DISCUSSION TITLE
How to run production build in development?

### DISCUSSION THREAD

**Discussion by samiemad** [excalidraw/excalidraw#5107] [Q&A]
So I'm trying to integrate excalidraw in our product. We are using React and importing the library as an ES6 module.

We have strict Content Security Policy (CSP). So allowing `unsafe-eval` is not an option for us.

As far as I can see, the development build of excalidraw is using `eval` (probably to inject source maps). The production build does not.

So if I manually edit `excalidraw/main.js` to use production build, everything works fine. but the change is not persisted

I am wondering if there is a more robust way to use the production build when importing as an ES6 module. Can you help?

Alternatively, removing `eval` from the development build should also help here. I don't know if that is an option.

Best,
Sami

**Comment by dwelle** [comment]
You'd have to set your `NODE_ENV` to `production` (see https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/main.js).

**Reply by samiemad** [reply]
Unfortunately we cannot do that. In some places in our code we use this variable to enable debugging features.

For now I found a workaround by directly using `@excalidraw/excalidraw/dist/excalidraw.production.min` instead of `@excalidraw/excalidraw`
And applying the types by creating a `d.ts` file with 
```
declare module '@excalidraw/excalidraw/dist/excalidraw.production.min' {
  import Excalidraw from '@excalidraw/excalidraw/types';
  export default Excalidraw;
}
```

Thanks for the help anyway. Cheers!
